reference_validator/hotvalidator.py

```diff
@@ -0,0 +1,751 @@
+                            if self.pretty_format:
+                                print('Parameter ' + enum.Fonts.YELLOW + ref.referent + enum.Fonts.DEFAULT +
+                                      ' has no corresponding default or property in ' +  enum.Fonts.YELLOW +
+                                      ref.element + enum.Fonts.DEFAULT + ' in ' +
+                                      enum.Fonts.YELLOW + os.path.relpath(ref.parent, self.init_dir) + enum.Fonts.DEFAULT + '.')
+                            else:
+                                print('Parameter ' + ref.referent + ' has no corresponding default or property in ' +
+                                      ref.element + ' in ' + os.path.relpath(ref.parent, self.init_dir) + '.')
+
+                        # missing parameter
+                        elif ref.type == enum.ErrorTypes.MISS_PARAM:
+                            if self.pretty_format:
+                                print('Property ' + enum.Fonts.YELLOW + ref.referent + enum.Fonts.DEFAULT +
+                                      ' has no corresponding parameter in ' + enum.Fonts.YELLOW +
+                                      os.path.relpath(ref.parent, self.init_dir) + enum.Fonts.DEFAULT + '.')
+                            else:
+                                print('Property ' + ref.referent + ' has no corresponding parameter in ' +
+                                      os.path.relpath(ref.parent, self.init_dir) + '.')
+
+                        # dependency not found
+                        elif ref.type == enum.ErrorTypes.DEPENDS_ON:
+                            if self.pretty_format:
+                                print('Resource ' + enum.Fonts.YELLOW + ref.referent + enum.Fonts.DEFAULT +
+                                      ' that resource ' +  enum.Fonts.YELLOW +
+                                      ref.element + enum.Fonts.DEFAULT + ' depends on is not declared.')
+                            else:
+                                print('Resource ' + ref.referent + ' that resource ' +
+                                      ref.element + ' depends on is not declared.')
+                    print('')
+
+                # Unused parameters
+                if False in [x.used for x in node.params]:
+                    if self.pretty_format:
+                        print(enum.Fonts.BOLD +  'Unused parameters:' + enum.Fonts.DEFAULT)
+                    else:
+                        print('Unused parameters:')
+
+                    for par in node.params:
+                        if par.used == False:
+                            if self.pretty_format:
+                                print('- ' + enum.Fonts.YELLOW + par.name + enum.Fonts.DEFAULT)
+                            else:
+                                print('- ' + par.name)
+                    print('')
+
+                # Print hidden parameters (optional)
+                if (self.print_unused) and [True for x in node.params if x.hidden]:
```

Comment:
I think this can be simplified into:

    if self.print_unused and any(x.hidden for x in node.params):